ocaml/testsuite/tests/typing-modes/class.ml

```diff
@@ -72,6 +72,19 @@ Line 4, characters 30-31:
 Error: This value is nonportable but expected to be portable.
 |}]
 
+(* values written to instance variables need to be legacy *)
+class cla = object
+    val mutable x = "hello"
+
+    method foo = x <- (world : _ @@ local)
```

Comment:
What’s `world`? Maybe it doesn’t matter for the test, but it’s a bit strange to use an out-of-scope variable here.

Reply:
Yeah it looks strange. I mean it kind of show that the mode is constrained before diving down the sub-expression. But yes it's confusing and fixed.